src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -1015,6 +1015,50 @@ void InstanceKlass::initialize_super_interfaces(TRAPS) {
   }
 }
 
+ResourceHashtable<const InstanceKlass*, OopHandle, 107, ResourceObj::C_HEAP, mtClass>
+      _initialization_error_table;
+
+void InstanceKlass::add_initialization_error(Handle exception, TRAPS) {
+  bool created = false;
+  // Create the same exception with a message indicating the thread name,
+  // and the StackTraceElements.
+  // If the initialization error is OOM, this might not work, but if GC kicks in
+  // this would be still be helpful.
+  Handle cause = java_lang_Throwable::get_cause_with_stack_trace(exception, THREAD);
+  CLEAR_PENDING_EXCEPTION;
```

Comment:
If new exceptions happened inside `get_cause_with_stack_trace`, `cause` will be null here. Maybe proceed to the following only if `cause` is not null?

Reply:
I thought of doing that also, but now can't remember why I didn't do it that way.  Fixed.